marklogic-data-hub-central/ui/src/components/entities/matching/matching-step-detail/matching-step-detail.tsx

```diff
@@ -159,6 +160,13 @@ const MatchingStepDetail: React.FC = () => {
     await test();
     setPreviewMatchedActivity(previewMatchActivity);
     setRulesetDataList(rulesetDataList);
+    let response = await axios.get("/api/steps/matching");
```

Comment:
Let's put all API calls under an appropriate location within **/src/api/***.
This seems like a good candidate for "getSteps", under **/src/api/steps.tsx** and then we can simply call it here like **getSteps("matching")**
Also, let's add another layer of validation by checking if you received a proper response before actually using it further. One of the ways is to check if the value of response.status is "200".